src/oscilloscope.cpp

```diff
@@ -4113,8 +4113,6 @@ void Oscilloscope::writeAllSettingsToHardware()
 
 		/*iio_device_attr_write_longlong(adc->iio_adc_dev(),
 			"oversampling_ratio", 1);*/
-
-		m2k_adc->setSampleRate(active_sample_rate);
```

Comment:
This line fixed a bug where the sample rate would not be set when disabling/enabling the filter and playing with the spectrum analyzer. Fix introduced in commit: https://github.com/analogdevicesinc/scopy/commit/2fc96ccec862f21d0706990a31a028e3ca544c42